### PR TITLE
Mark advanced search tests as affected with bug 1283554

### PR DESCRIPTION
**Pull request by vrutkovs**
This also fixes several utf-8 decode problems in logging - this is required for other advanced search tests to pass

{{pytest: -v --long-running cfme/tests/infrastructure/test_advanced_search_host.py }}

### CODE REVIEW

**Comment by dajoRH** [comment]
Lint report for commit b9d4dfa89bba20111d67f9eb1416fb83eb506e89:
:godmode: All seems good! :cake: :punch: :cookie: 
*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit dbd0a50f115cf01bda1b50804a2198a95e9c17eb:
:godmode: All seems good! :cake: :punch: :cookie: 
*CFME QE Bot*

File: fixtures/browser.py [hunk 1]
```diff
@@ -29,7 +29,7 @@ def pytest_runtest_setup(item):
 
 def pytest_exception_interact(node, call, report):
     name, location = get_test_idents(node)
-    val = unicode(call.excinfo.value)
+    val = unicode(call.excinfo.value.message.decode('utf-8'))
```
Comment (mfalesni) [inline fixtures/browser.py]: In this case, the `unicode()` is unnecessary, product of decode is unicode :)

File: fixtures/browser.py [hunk 2]
```diff
@@ -29,7 +29,7 @@ def pytest_runtest_setup(item):
...
     short_tb = '%s\n%s' % (call.excinfo.type.__name__, val.encode('ascii', 'ignore'))
```
Comment (mfalesni) [inline fixtures/browser.py]: And while here, can you put `"xmlcharrefreplace"` instead of ignore?

File: utils/log.py [hunk 3]
```diff
@@ -476,7 +476,7 @@ def art_log(self, level_name, message, kwargs):
             message = str(message)
         art_log_record = {
             'level': level_name,
-            'message': message.encode("ascii", "xmlcharrefreplace"),
+            'message': message.decode('utf-8').encode("ascii", "xmlcharrefreplace"),
```
Comment (mfalesni) [inline utils/log.py]: For this, I think the correct approach would be:
* If the message is not a basestring, inspect it if it has `__unicode__`. If yes, do  `unicode()` on it and continue with step 2. If it does not have `__unicode__`, do `str()` on it and continue with step 3.
* check if message is unicode, if yes, encode with xmlcharrefreplace
* if message is a str, do `.decode("utf-8", "ignore").encode("ascii", "xmlcharrefreplace")`

This should ensure proper handling of any string, unicode or any object passed to the function.

@psav ?

File: fixtures/browser.py (hunk 1 above)
Comment (mfalesni) [inline fixtures/browser.py]: and maybe keep the `"ignore"` param in it, just to play safe, away from the various UnicodeDecodeErrors and similar pleasant stuff.

File: utils/log.py (hunk 3 above)
Comment (psav) [inline utils/log.py]: That flow seems to be good, we should functionize it and put it in utils maybe?

File: fixtures/browser.py (hunk 1 above)
Comment (psav) [inline fixtures/browser.py]: Agreed

File: utils/log.py (hunk 3 above)
Comment (mfalesni) [inline utils/log.py]: something like `digest_string` ?

**Comment by dajoRH** [comment]
Lint report for commit 68f83abc1a220839765b980bf1aba0f9cecdc13e:
:godmode: All seems good! :cake: :punch: :cookie: 
*CFME QE Bot*

File: utils/log.py (hunk 3 above)
Comment (vrutkovs) [inline utils/log.py]: Rewrote all messages to be first decoded (with 'ignore') and then encoded (with xmlcharref).

I still suspect there is a better, out-of-the-box solution for this, lets see if it helps with those crashes

**Pull request merged**